subtree-of-another-tree/jdalma.kt

```diff
@@ -0,0 +1,28 @@
+package leetcode_study
+
+import io.kotest.matchers.shouldBe
+import org.junit.jupiter.api.Test
+
+class `subtree-of-another-tree` {
+
+    fun isSubtree(root: TreeNode?, subRoot: TreeNode?): Boolean {
+        if (root == null) return false
+        else if (isSame(root, subRoot)) return true
+
+        return isSubtree(root.left, subRoot) || isSubtree(root.right, subRoot)
+    }
+
+    private fun isSame(root: TreeNode?, subRoot: TreeNode?): Boolean {
+        if (root == null && subRoot == null) return true
+        else if (root == null || subRoot == null) return false
+        else if (root.`val` != subRoot.`val`) return false
```

Comment:
불필요한 `else`를 제거하면 코드가 좀 더 정갈해 보이지 않을까요?

```suggestion
        if (root == null && subRoot == null) return true
        if (root == null || subRoot == null) return false
        if (root.`val` != subRoot.`val`) return false
```